src/trunnel/netinfo.trunnel

```diff
@@ -5,14 +5,14 @@ const NETINFO_ADDR_TYPE_IPV4 = 4;
 const NETINFO_ADDR_TYPE_IPV6 = 6;
 
 struct netinfo_addr {
-  u8 addr_type IN [NETINFO_ADDR_TYPE_IPV4, NETINFO_ADDR_TYPE_IPV6];
-  u8 len IN [4, 16];
+  u8 addr_type;
+  u8 len;
   union addr[addr_type] {
     NETINFO_ADDR_TYPE_IPV4: u32 ipv4;
     NETINFO_ADDR_TYPE_IPV6: u8 ipv6[16];
     default: fail;
```

Comment:
This is still going to mean that if addr_type isn't recognized, parsing fails.

Try replacing the union line with `union addr[addr_type] with length len {`, and the default line with `default: ignore` -- I think that will do it.

Also, we should have unit tests that make sure that this case actually works.

Reply:
Fixed netinfo.trunnel in c92c0cbc9f7a1ad889cfdda6943ca484d110438e. Added unit test in c659603ac5a428d93ea625aa28b6b51ef9f2f9b3.